File: translation_output/wizardcoder/codenet/java/python-sanitized/p03999.py
import java.util.*;

class p03999:
    def main(self, args):
        sc = Scanner(System.in)
        s = sc.next()
        ans = 0
        for dx in range(s.length()):
            for start in range(s.length()-dx):
                cur = int(s[start:start+dx+1])
                ans += cur*(self.helper(start)*self.helper(s.length()-1-dx-start))
        print(ans)

    def helper(self, n):
        if n == 0:
            return 1
        return 1<<(n-1)

File: Results/DER/Translation/translation_output/no_test/granite/codenet/python/java-sanitized/p03999.java
import java.util.Scanner;

public class p03999 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        int ans = 0;
        for (int i = 0; i < Math.pow(2, s.length() - 1); i++) {
            String pat = s.charAt(0) + "";
            int index = 1;
            for (int j = 0; j < s.length() - 1; j++) {
                if ((i >> j) % 2 == 1) {
                    pat += "+";
                }
                pat += s.charAt(index);
                index++;
            }
            //System.out.println(pat);
            ans += eval(pat);
        }
        System.out.println(ans);
    }

    public static int eval(String s) {
        int ans = 0;
        int num = 0;
        char op = '+';
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '+' || s.charAt(i) == '-') {
                ans += num;
                num = 0;
                op = s.charAt(i);
            } else if (s.charAt(i) == '(') {
                int start = i;
                int end = s.indexOf(')', i);
                int sub = eval(s.substring(start + 1, end));
                if (op == '+') {
                    ans += sub;
                } else {
                    ans -= sub;
                }
                i = end;
            } else {
                num = num * 10 + s.charAt(i) - '0';
            }
        }
        ans += num;
        return ans;
    }
}